### PR TITLE
r/storagegateway_nfs_file_share - support `file_share_name`

### PR DESCRIPTION
**Pull request by DrFaust92**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
resource_aws_storagegateway_nfs_file_share - support `file_share_name`
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSStorageGatewayNfsFileShare_'
--- PASS: TestAccAWSStorageGatewayNfsFileShare_basic (286.69s)
--- PASS: TestAccAWSStorageGatewayNfsFileShare_disappears (314.00s)
--- PASS: TestAccAWSStorageGatewayNfsFileShare_cacheAttributes (545.97s)
--- PASS: TestAccAWSStorageGatewayNfsFileShare_KMSKeyArn (554.94s)
--- PASS: TestAccAWSStorageGatewayNfsFileShare_ReadOnly (409.35s)
--- PASS: TestAccAWSStorageGatewayNfsFileShare_Squash (459.57s)
--- PASS: TestAccAWSStorageGatewayNfsFileShare_ObjectACL (377.12s)
--- PASS: TestAccAWSStorageGatewayNfsFileShare_RequesterPays (393.68s)
--- PASS: TestAccAWSStorageGatewayNfsFileShare_KMSEncrypted (341.95s)
--- PASS: TestAccAWSStorageGatewayNfsFileShare_DefaultStorageClass (444.18s)
--- PASS: TestAccAWSStorageGatewayNfsFileShare_fileShareName (428.53s)
--- PASS: TestAccAWSStorageGatewayNfsFileShare_NFSFileShareDefaults (469.43s)
--- PASS: TestAccAWSStorageGatewayNfsFileShare_tags (543.66s)
--- PASS: TestAccAWSStorageGatewayNfsFileShare_ClientList (672.34s)
--- PASS: TestAccAWSStorageGatewayNfsFileShare_GuessMIMETypeEnabled (444.01s)
```

### CODE REVIEW

**Comment by ewbankkit** [comment]
Relates: https://github.com/hashicorp/terraform-provider-aws/pull/16008.

File: website/docs/r/storagegateway_nfs_file_share.html.markdown [hunk 1]
```diff
@@ -39,6 +39,7 @@ The following arguments are supported:
 * `read_only` - (Optional) Boolean to indicate write status of file share. File share does not accept writes if `true`. Defaults to `false`.
 * `requester_pays` - (Optional) Boolean who pays the cost of the request and the data download from the Amazon S3 bucket. Set this value to `true` if you want the requester to pay instead of the bucket owner. Defaults to `false`.
 * `squash` - (Optional) Maps a user to anonymous user. Defaults to `RootSquash`. Valid values: `RootSquash` (only root is mapped to anonymous user), `NoSquash` (no one is mapped to anonymous user), `AllSquash` (everyone is mapped to anonymous user)
+* `file_share_name` - (Optional) The name of the file share. must be set if an S3 prefix name is set in `location_arn`.
```
Comment (ewbankkit) [inline website/docs/r/storagegateway_nfs_file_share.html.markdown:42]:
```suggestion
* `file_share_name` - (Optional) The name of the file share. Must be set if an S3 prefix name is set in `location_arn`.
```

**Review by ewbankkit — APPROVED** [review]
LGTM.

```console
$ make testacc TEST=./aws TESTARGS='-run=TestAccAWSStorageGatewayNfsFileShare_fileShareName'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSStorageGatewayNfsFileShare_fileShareName -timeout 120m
=== RUN   TestAccAWSStorageGatewayNfsFileShare_fileShareName
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_fileShareName
=== CONT  TestAccAWSStorageGatewayNfsFileShare_fileShareName
--- PASS: TestAccAWSStorageGatewayNfsFileShare_fileShareName (284.18s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	284.238s
```

**Comment by breathingdust** [comment]
LGTM :rocket: Thanks @DrFaust92 for the contribution!

Verified Acceptance Tests in Commercial (us-west-2)

```console
make testacc TEST=./aws TESTARGS='-run=TestAccAWSStorageGatewayNfsFileShare_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSStorageGatewayNfsFileShare_ -timeout 120m
=== RUN   TestAccAWSStorageGatewayNfsFileShare_basic
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_basic
=== RUN   TestAccAWSStorageGatewayNfsFileShare_tags
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_tags
=== RUN   TestAccAWSStorageGatewayNfsFileShare_fileShareName
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_fileShareName
=== RUN   TestAccAWSStorageGatewayNfsFileShare_ClientList
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_ClientList
=== RUN   TestAccAWSStorageGatewayNfsFileShare_DefaultStorageClass
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_DefaultStorageClass
=== RUN   TestAccAWSStorageGatewayNfsFileShare_GuessMIMETypeEnabled
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_GuessMIMETypeEnabled
=== RUN   TestAccAWSStorageGatewayNfsFileShare_KMSEncrypted
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_KMSEncrypted
=== RUN   TestAccAWSStorageGatewayNfsFileShare_KMSKeyArn
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_KMSKeyArn
=== RUN   TestAccAWSStorageGatewayNfsFileShare_NFSFileShareDefaults
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_NFSFileShareDefaults
=== RUN   TestAccAWSStorageGatewayNfsFileShare_ObjectACL
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_ObjectACL
=== RUN   TestAccAWSStorageGatewayNfsFileShare_ReadOnly
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_ReadOnly
=== RUN   TestAccAWSStorageGatewayNfsFileShare_RequesterPays
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_RequesterPays
=== RUN   TestAccAWSStorageGatewayNfsFileShare_Squash
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_Squash
=== RUN   TestAccAWSStorageGatewayNfsFileShare_cacheAttributes
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_cacheAttributes
=== RUN   TestAccAWSStorageGatewayNfsFileShare_disappears
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_disappears
=== CONT  TestAccAWSStorageGatewayNfsFileShare_basic
=== CONT  TestAccAWSStorageGatewayNfsFileShare_NFSFileShareDefaults
=== CONT  TestAccAWSStorageGatewayNfsFileShare_cacheAttributes
=== CONT  TestAccAWSStorageGatewayNfsFileShare_DefaultStorageClass
=== CONT  TestAccAWSStorageGatewayNfsFileShare_KMSKeyArn
=== CONT  TestAccAWSStorageGatewayNfsFileShare_KMSEncrypted
=== CONT  TestAccAWSStorageGatewayNfsFileShare_Squash
=== CONT  TestAccAWSStorageGatewayNfsFileShare_GuessMIMETypeEnabled
=== CONT  TestAccAWSStorageGatewayNfsFileShare_fileShareName
=== CONT  TestAccAWSStorageGatewayNfsFileShare_disappears
=== CONT  TestAccAWSStorageGatewayNfsFileShare_ClientList
=== CONT  TestAccAWSStorageGatewayNfsFileShare_RequesterPays
=== CONT  TestAccAWSStorageGatewayNfsFileShare_ObjectACL
=== CONT  TestAccAWSStorageGatewayNfsFileShare_ReadOnly
=== CONT  TestAccAWSStorageGatewayNfsFileShare_tags
2020/11/17 14:35:16 [WARN] Truncating attribute path of 0 diagnostics for TypeSet
2020/11/17 14:35:16 [WARN] Truncating attribute path of 0 diagnostics for TypeSet
2020/11/17 14:35:16 [WARN] Truncating attribute path of 0 diagnostics for TypeSet
2020/11/17 14:35:21 [DEBUG] Waiting for state to become: [success]
2020/11/17 14:36:24 [INFO] AWS Auth provider used: "SharedCredentialsProvider"
--- PASS: TestAccAWSStorageGatewayNfsFileShare_disappears (233.37s)
--- PASS: TestAccAWSStorageGatewayNfsFileShare_KMSEncrypted (239.53s)
--- PASS: TestAccAWSStorageGatewayNfsFileShare_basic (258.49s)
--- PASS: TestAccAWSStorageGatewayNfsFileShare_fileShareName (285.94s)
--- PASS: TestAccAWSStorageGatewayNfsFileShare_Squash (289.51s)
--- PASS: TestAccAWSStorageGatewayNfsFileShare_ReadOnly (293.96s)
--- PASS: TestAccAWSStorageGatewayNfsFileShare_DefaultStorageClass (298.18s)
--- PASS: TestAccAWSStorageGatewayNfsFileShare_NFSFileShareDefaults (298.33s)
--- PASS: TestAccAWSStorageGatewayNfsFileShare_ObjectACL (305.71s)
--- PASS: TestAccAWSStorageGatewayNfsFileShare_tags (307.41s)
--- PASS: TestAccAWSStorageGatewayNfsFileShare_GuessMIMETypeEnabled (336.60s)
--- PASS: TestAccAWSStorageGatewayNfsFileShare_cacheAttributes (337.57s)
--- PASS: TestAccAWSStorageGatewayNfsFileShare_RequesterPays (348.27s)
--- PASS: TestAccAWSStorageGatewayNfsFileShare_ClientList (387.98s)
--- PASS: TestAccAWSStorageGatewayNfsFileShare_KMSKeyArn (450.87s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	452.792s
```

Verified Acceptance Tests in GovCloud (us-gov-west-1)

```console
make testacc TEST=./aws TESTARGS='-run=TestAccAWSStorageGatewayNfsFileShare_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSStorageGatewayNfsFileShare_ -timeout 120m
=== RUN   TestAccAWSStorageGatewayNfsFileShare_basic
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_basic
=== RUN   TestAccAWSStorageGatewayNfsFileShare_tags
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_tags
=== RUN   TestAccAWSStorageGatewayNfsFileShare_fileShareName
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_fileShareName
=== RUN   TestAccAWSStorageGatewayNfsFileShare_ClientList
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_ClientList
=== RUN   TestAccAWSStorageGatewayNfsFileShare_DefaultStorageClass
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_DefaultStorageClass
=== RUN   TestAccAWSStorageGatewayNfsFileShare_GuessMIMETypeEnabled
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_GuessMIMETypeEnabled
=== RUN   TestAccAWSStorageGatewayNfsFileShare_KMSEncrypted
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_KMSEncrypted
=== RUN   TestAccAWSStorageGatewayNfsFileShare_KMSKeyArn
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_KMSKeyArn
=== RUN   TestAccAWSStorageGatewayNfsFileShare_NFSFileShareDefaults
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_NFSFileShareDefaults
=== RUN   TestAccAWSStorageGatewayNfsFileShare_ObjectACL
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_ObjectACL
=== RUN   TestAccAWSStorageGatewayNfsFileShare_ReadOnly
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_ReadOnly
=== RUN   TestAccAWSStorageGatewayNfsFileShare_RequesterPays
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_RequesterPays
=== RUN   TestAccAWSStorageGatewayNfsFileShare_Squash
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_Squash
=== RUN   TestAccAWSStorageGatewayNfsFileShare_cacheAttributes
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_cacheAttributes
=== RUN   TestAccAWSStorageGatewayNfsFileShare_disappears
=== PAUSE TestAccAWSStorageGatewayNfsFileShare_disappears
=== CONT  TestAccAWSStorageGatewayNfsFileShare_basic
=== CONT  TestAccAWSStorageGatewayNfsFileShare_NFSFileShareDefaults
=== CONT  TestAccAWSStorageGatewayNfsFileShare_DefaultStorageClass
=== CONT  TestAccAWSStorageGatewayNfsFileShare_tags
=== CONT  TestAccAWSStorageGatewayNfsFileShare_cacheAttributes
=== CONT  TestAccAWSStorageGatewayNfsFileShare_Squash
=== CONT  TestAccAWSStorageGatewayNfsFileShare_RequesterPays
=== CONT  TestAccAWSStorageGatewayNfsFileShare_ReadOnly
=== CONT  TestAccAWSStorageGatewayNfsFileShare_ObjectACL
=== CONT  TestAccAWSStorageGatewayNfsFileShare_KMSEncrypted
=== CONT  TestAccAWSStorageGatewayNfsFileShare_fileShareName
=== CONT  TestAccAWSStorageGatewayNfsFileShare_KMSKeyArn
=== CONT  TestAccAWSStorageGatewayNfsFileShare_ClientList
=== CONT  TestAccAWSStorageGatewayNfsFileShare_GuessMIMETypeEnabled
=== CONT  TestAccAWSStorageGatewayNfsFileShare_disappears
2020/11/17 14:41:30 [INFO] AWS Auth provider used: "SharedCredentialsProvider"
2020/11/17 14:41:30 [WARN] Truncating attribute path of 0 diagnostics for TypeSet
2020/11/17 14:41:30 [DEBUG] Waiting for state to become: [success]
2020/11/17 14:41:30 [DEBUG] Waiting for state to become: [success]
2020/11/17 14:41:30 [DEBUG] setting computed for "versioning" from ComputedKeys
2020/11/17 14:41:30 [DEBUG] S3 bucket create: tf-acc-test-306670832944446823
2020/11/17 14:41:30 [DEBUG] S3 bucket tf-acc-test-306670832944446823 has canned ACL private
2020/11/17 14:41:30 [DEBUG] S3 bucket create: tf-acc-test-306670832944446823, using region: us-gov-west-1
2020/11/17 14:41:30 [DEBUG] Waiting for state to become: [success]
2020/11/17 14:41:30 [DEBUG] Trying to create new S3 bucket: "tf-acc-test-306670832944446823"
2020/11/17 14:43:55 [DEBUG] Reading Storage Gateway SMB Settings: {
  GatewayARN: "arn:aws-us-gov:storagegateway:us-gov-west-1:357342307427:gateway/sgw-978066FE"
}
2020/11/17 14:44:01 [WARN] Truncating attribute path of 0 diagnostics for TypeSet
2020/11/17 14:44:01 [WARN] Truncating attribute path of 0 diagnostics for TypeSet
2020/11/17 14:44:32 [INFO] AWS Auth provider used: "SharedCredentialsProvider"
--- PASS: TestAccAWSStorageGatewayNfsFileShare_disappears (203.21s)
--- PASS: TestAccAWSStorageGatewayNfsFileShare_KMSEncrypted (212.57s)
--- PASS: TestAccAWSStorageGatewayNfsFileShare_basic (240.76s)
--- PASS: TestAccAWSStorageGatewayNfsFileShare_tags (250.82s)
--- PASS: TestAccAWSStorageGatewayNfsFileShare_ReadOnly (264.13s)
--- PASS: TestAccAWSStorageGatewayNfsFileShare_RequesterPays (267.59s)
--- PASS: TestAccAWSStorageGatewayNfsFileShare_GuessMIMETypeEnabled (274.54s)
--- PASS: TestAccAWSStorageGatewayNfsFileShare_Squash (307.63s)
--- PASS: TestAccAWSStorageGatewayNfsFileShare_NFSFileShareDefaults (308.28s)
--- PASS: TestAccAWSStorageGatewayNfsFileShare_DefaultStorageClass (315.51s)
--- PASS: TestAccAWSStorageGatewayNfsFileShare_fileShareName (317.54s)
--- PASS: TestAccAWSStorageGatewayNfsFileShare_ClientList (320.33s)
--- PASS: TestAccAWSStorageGatewayNfsFileShare_ObjectACL (326.75s)
--- PASS: TestAccAWSStorageGatewayNfsFileShare_KMSKeyArn (344.21s)
--- PASS: TestAccAWSStorageGatewayNfsFileShare_cacheAttributes (384.61s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	386.095s
```

**Pull request merged**